### PR TITLE
CloudCare Synchronous Case Claim

### PR DESCRIPTION
**Pull request by wpride**
Cross https://github.com/dimagi/commcare-hq/pull/12800

### CODE REVIEW

**Comment by czue** [comment]
hiyooo

File: src/main/java/application/AbstractBaseController.java [hunk 1]
```diff
@@ -54,33 +55,35 @@
     private final Log log = LogFactory.getLog(AbstractBaseController.class);
 
 
-    public Object resolveFormGetNext(MenuSession menuSession) throws Exception {
+    public BaseResponseBean resolveFormGetNext(MenuSession menuSession) throws Exception {
         menuSession.getSessionWrapper().syncState();
         if(menuSession.getSessionWrapper().finishExecuteAndPop(menuSession.getSessionWrapper().getEvaluationContext())){
-            Object nextMenu = getNextMenu(menuSession);
+            BaseResponseBean nextMenu = getNextMenu(menuSession);
```
Comment (benrudolph) [inline src/main/java/application/AbstractBaseController.java]: 😄 thanks

File: src/main/java/screens/FormplayerSyncScreen.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            HttpEntity<String> entity = new HttpEntity<>(builtQuery, headers);
+            RestTemplate restTemplate = new RestTemplate();
+            ResponseEntity<String> response =
+                    restTemplate.exchange(syncPost.getUrl().toString(),
+                            HttpMethod.POST,
+                            entity, String.class);
+            return response;
+        } else {
+            // expected a sync entry; clear session and show vague 'session error' message to user
+            //clearSessionAndExit(asw, true);
```
Comment (benrudolph) [inline src/main/java/screens/FormplayerSyncScreen.java]: was this purposeful?

**Comment by benrudolph** [comment]
conflicts, but looking good! 💼 🔗 🔗 🔗 📱

File: src/main/java/screens/FormplayerSyncScreen.java (hunk 2 above)
Comment (wpride) [inline src/main/java/screens/FormplayerSyncScreen.java]: good catch, added error handling

**Pull request merged**